### PR TITLE
RAC-5940 Refactor/build ova sh

### PR DESCRIPTION
**Pull request by changev**
extract build_ova.sh from HWIO-BUILD
@panpan0000 @PengTian0 

waiting for https://github.com/RackHD/RackHD/pull/875 to be merged

### CODE REVIEW

**Comment by panpan0000** [comment]
Please review your code, your PR includes other people's commit
-1

File: packer/build_ova.sh [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+  echo "Function: this script is used to build rackhd ova":
+  echo "usage: $0 [arguments]"
+  echo "  Mandatory Arguments:"
+  echo "    --RACKHD_DIR: The directory of rackhd repo."
+  echo "  Optional Arguments:"
+  echo "    --OS_VER: The ubuntu iso version for ova build.
+                      Default: ubuntu-14.04"
+  echo "    --ANSIBLE_PLAYBOOK: Use which playbook to install rackhd and dependency.
+                                Default: rackhd_local.yml"
+  echo "    --DEBIAN_REPOSITORY: by defaul is to using RackHD Bintray repo: deb https://dl.bintray.com/rackhd/debian trusty release,
```
Comment (panpan0000) [inline packer/build_ova.sh]: defaul --> default

File: packer/build_ova.sh [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+                                 if using latest nightly build, can choose deb https://dl.bintray.com/rackhd/debian trusty main.
+                                 and other repo like private artifactory is also ok. just let the /etc/apt/source.list recognize it."
+  echo "    --RACKHD_VERSION: The version of rackhd that will be installed in to ova.
```
Comment (panpan0000) [inline packer/build_ova.sh]: can you be more specific?
```the version of rackhd debian package that will be installed into ova.```

File: packer/build_ova.sh [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+           echo "${WARNING_HEADER} Skip the ENV VAR: RACKHD_VERSION=$RACKHD_VERSION, this variable is useless for rackhd_local.yml (build from src code)".
+       fi
+    else
+       if [ $ANSIBLE_PLAYBOOK != "rackhd_local" ]; then
+           echo "${INFO_HEADER} Install RackHD Package without specific version(default latest)."
+       fi
+    fi
+
+    if [  -n "${DEBIAN_REPOSITORY}" ];  then
+        if [ $ANSIBLE_PLAYBOOK != "rackhd_local" ]; then
```
Comment (panpan0000) [inline packer/build_ova.sh]: should be ``` if [ $ANSIBLE_PLAYBOOK = "rackhd_local" ]; then```

File: packer/build_ova.sh [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+            echo "${WARNING_HEADER} Skip the ENV VAR: DEBIAN_REPOSITORY=$DEBIAN_REPOSITORY."
+        else
+            echo "${WARNING_HEADER} DEBIAN_REPOSITORY variable is useless for rackhd_local.yml (build from src code)".
```
Comment (panpan0000) [inline packer/build_ova.sh]: should be {INFO_HEADER} and missing a $ before DEBIAN_REPOSITORY

File: packer/build_ova.sh [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+
+    # execute 'packer build'
+    if [ "$BUILD_STAGE" == "BUILD_TEMPLATE" ]; then
+        #Build from  iso, create a Pre-build Cache ( vmware:iso-->vmx, virtualbox: iso-->ovf )
+        $PACKER build --force --only=vmware-iso  --var-file=$CFG_FILE  ${PACKER_TEMP}  | tee packer-install.log
+    else
+        if [ "$BUILD_STAGE" == "BUILD_FINAL" ]; then
+             #Build from cache template, and output the final image .(vmware: vmx-->ova, virtualbox: ovf-->box)
+             mkdir output-vmware-iso && mv $CACHE_IMAGE_DIR/* ./output-vmware-iso
+             $PACKER build --force --only=vmware-vmx  --var-file=$CFG_FILE  --var "playbook=${ANSIBLE_PLAYBOOK}_mini" ${PACKER_TEMP}   | tee packer-install.log
```
Comment (panpan0000) [inline packer/build_ova.sh]: that was my bad , to add tricky ```_mini``` before..
so can we directly use the passing in ANSIBLE_PLAYBOOK without add a surfix  ?
and don't forget to change where build_ova.sh is invoked from ```rackhd_package``` to ```rackhd_package_mini```

File: packer/build_ova.sh [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+        fi
+    fi
+    ## Check packer build command status, because above we use pipeline, so use  ${PIPESTATUS[0]} to catch return-code of command before pipeline.(only works on bash)
+    if [ ${PIPESTATUS[0]} != 0 ]; then
+        echo "${ERROR_HEADER} Packer Build failed.. exit"
+        exit 3
+    fi
+    if  [ "$BUILD_STAGE" == "BUILD_TEMPLATE" ]; then
+         echo "{INFO_HEADER} Build VMX is successful... Exit the HWIMO-BUILD Script. Use BUILD_STAGE=BUILD_FINAL and ANSIBLE_PLAYBOOK=rackhd_mini_xxxx env variable to build OVA from this VMX."
```
Comment (PengTian0) [inline packer/build_ova.sh]: HWIMO-BUILD?

File: packer/build_ova.sh [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+}
+
+packerBuildOVA(){
+    #packer template file
+    PACKER_TEMP=template-${OS_VER}.json
+
+    # execute 'packer build'
+    if [ "$BUILD_STAGE" == "BUILD_TEMPLATE" ]; then
+        #Build from  iso, create a Pre-build Cache ( vmware:iso-->vmx, virtualbox: iso-->ovf )
+        $PACKER build --force --only=vmware-iso  --var-file=$CFG_FILE  ${PACKER_TEMP}  | tee packer-install.log
```
Comment (PengTian0) [inline packer/build_ova.sh]: Maybe there should be another shell to build vmx.
And another shell script to build ovf and I will do that.

File: packer/build_ova.sh (hunk 6 above)
Comment (PengTian0) [reply packer/build_ova.sh]: The script doesn't exit here. It will continue to run next method and fail.

File: packer/build_ova.sh (hunk 4 above)
Comment (changev) [reply packer/build_ova.sh]: 1. I think this should be warning. User use rackhd local and use deb repo too. Maybe the user use the wrong yml file. So warning can help user to check this.
2. DEBIAN_REPOSITORY is var name not var value here.

**Comment by changev** [comment]
@panpan0000  Please check the latest code

File: packer/build_ova.sh [hunk 8]
```diff
@@ -0,0 +1,339 @@
+#!/bin/bash
+set -x
```
Comment (panpan0000) [inline packer/build_ova.sh]: you guys always like "set -x"  -----cry to tear.......

**Comment by panpan0000** [comment]
RAC-5940

**Comment by changev** [comment]
test this please

**Pull request merged**